executor/executor.go

```diff
@@ -117,13 +122,15 @@ func (e *baseExecutor) Schema() *expression.Schema {
 }
 
 // newFirstChunk creates a new chunk to buffer current executor's result.
-func (e *baseExecutor) newFirstChunk() *chunk.Chunk {
-	return chunk.New(e.retTypes(), e.initCap, e.maxChunkSize)
+func newFirstChunk(e Executor) *chunk.Chunk {
+	base := e.This()
+	return chunk.New(base.retFieldTypes, base.initCap, base.maxChunkSize)
```

Comment:
@qw4990 @lysu I think we don't need initial capacity anymore? Can we remove this mechanism?

Reply:
yes